BCP195bis/draft-ietf-uta-rfc7525bis.md

```diff
@@ -587,21 +587,42 @@ are referred to Sec. 9.1 of {{RFC8446}} for cipher suite recommendations.
 
 ## Limits on Key Usage
 
-All ciphers have an upper limit on the amount of traffic that can be securely encrypted
-with any given key. In the case of AEAD cipher suites, the limit is typically determined
-by the cipher's integrity guarantees. When the amount of traffic for a particular connection
-has reached the limit, an implementation SHOULD perform a new handshake (or in TLS 1.3,
-a Key Update) to rotate the session key.
+All ciphers have an upper limit on the amount of traffic that can be securely
+protected with any given key. In the case of AEAD cipher suites, two separate
+limits are maintained for each key:
 
-For all AES-GCM cipher suites recommended for TLS 1.2 in this document, the limit
-for one connection is 2<sup>24.5</sup> full-size records (about 24 million).
-This is the same number as for TLS 1.3 with the equivalent cipher suites.
+1. Confidentiality limit (CL), i.e., the amount of records that can be
+   encrypted.
+1. Integrity limit (IL), i.e., the amount of records that can fail
+   authentication.
 
-<cref>TODO: refer to {{?I-D.irtf-cfrg-aead-limits}} once it has added the derivation
-for TLS 1.2, which is different from TLS 1.3. Different derivation, same numbers.</cref>
+The latter only applies to DTLS since TLS connections are torn down on the
+first decryption failure.
+
+When a sender is approaching CL, the implementation SHOULD initiate a new
+handshake (or in TLS 1.3, a Key Update) to rotate the session key.
+
+When a receiver has reached IL, the implementation SHOULD close the connection.
+When a receiver is approaching IL, the implementation SHOULD initiate a new
```

Comment:
Formally this is correct. But I don't think we should recommend re-handshake here: you've just received a quarter billion corrupted packets, just close the connection and be done.

Reply:
I think we should not shift from https://www.ietf.org/archive/id/draft-ietf-tls-dtls13-43.html#section-4.5.3-4
(I was with you before re-reading that para.)

Reply:
You are right, in fact it says you SHOULD drop it.  What it also says is that when approaching the limit you SHOULD start negotiating a new connection.  Is this second requirement that you don't want to mirror?

Reply:
Yes. I think any sane implementation would have timed out long before, because IL >> CL.

Reply:
Removing the second requirement is fine with me.